input_network_encoder.cpp

```diff
@@ -0,0 +1,371 @@
+					break;
+				case DataBuffer::DATA_TYPE_VECTOR2:
+					are_equals = SceneSynchronizer::compare(p_buffer_A.read_vector2(info.compression_level), p_buffer_B.read_vector2(info.compression_level), info.comparison_floating_point_precision);
+					break;
+				case DataBuffer::DATA_TYPE_NORMALIZED_VECTOR2:
+					are_equals = SceneSynchronizer::compare(p_buffer_A.read_normalized_vector2(info.compression_level), p_buffer_B.read_normalized_vector2(info.compression_level), info.comparison_floating_point_precision);
+					break;
+				case DataBuffer::DATA_TYPE_VECTOR3:
+					are_equals = SceneSynchronizer::compare(p_buffer_A.read_vector3(info.compression_level), p_buffer_B.read_vector3(info.compression_level), info.comparison_floating_point_precision);
+					break;
+				case DataBuffer::DATA_TYPE_NORMALIZED_VECTOR3:
+					are_equals = SceneSynchronizer::compare(p_buffer_A.read_normalized_vector3(info.compression_level), p_buffer_B.read_normalized_vector3(info.compression_level), info.comparison_floating_point_precision);
+					break;
+				case DataBuffer::DATA_TYPE_VARIANT:
+					are_equals = SceneSynchronizer::compare(p_buffer_A.read_variant(), p_buffer_B.read_variant(), info.comparison_floating_point_precision);
+					break;
+			};
+		}
+
+		if (!are_equals) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+uint32_t InputNetworkEncoder::count_size(DataBuffer &p_buffer) const {
+	int size = 0;
+	for (uint32_t i = 0; i < input_info.size(); i += 1) {
+		const NetworkedInputInfo &info = input_info[i];
+
+		const bool is_bool = info.default_value.get_type() == Variant::BOOL;
+
+		if (is_bool == false) {
+			size += p_buffer.read_bool_size();
+		}
+
+		switch (info.data_type) {
+			case DataBuffer::DATA_TYPE_BOOL:
+				size += p_buffer.read_bool_size();
+				break;
+			case DataBuffer::DATA_TYPE_INT:
+				size += p_buffer.read_int_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_REAL:
+				size += p_buffer.read_real_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_POSITIVE_UNIT_REAL:
+				size += p_buffer.read_positive_unit_real_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_UNIT_REAL:
+				size += p_buffer.read_unit_real_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_VECTOR2:
+				size += p_buffer.read_vector2_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_NORMALIZED_VECTOR2:
+				size += p_buffer.read_normalized_vector2_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_VECTOR3:
+				size += p_buffer.read_vector3_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_NORMALIZED_VECTOR3:
+				size += p_buffer.read_normalized_vector3_size(info.compression_level);
+				break;
+			case DataBuffer::DATA_TYPE_VARIANT:
+				size += p_buffer.read_variant_size();
+				break;
+		};
+	}
+
+	return size;
+}
+
+void InputNetworkEncoder::script_encode(const Array &p_inputs, Object *r_buffer) const {
```

Comment:
I dont think NetworkedInputInfo is exposed to GdScript, so what's expected as input here?

Reply:
The input is just an array of variant. Each variant holds the input value